src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  let stateClone = { ...state };
+
+  for (const { type, extraData, keysToRemove } of actions) {
+    let newStateClone = { ...stateClone };
```

Comment:
No need copy already copied state)
You can simple modify your `stateClone` object.